language/testing-infra/diem-transactional-test-runner/src/diem_test_harness.rs

```diff
@@ -0,0 +1,320 @@
+struct DiemRunArgs {
+    #[structopt(short = "k", long = "private-key", parse(try_from_str = parse_ed25519_private_key))]
+    privkey: Ed25519PrivateKey,
+}
+
+impl<'a> MoveTestAdapter<'a> for DiemTestAdapter<'a> {
+    type ExtraInitArgs = EmptyCommand;
+    type ExtraPublishArgs = DiemPublishArgs;
+    type ExtraRunArgs = DiemRunArgs;
+    type Subcommand = EmptyCommand;
+
+    fn compiled_state(&mut self) -> &mut CompiledState<'a> {
+        &mut self.compiled_state
+    }
+
+    fn default_syntax(&self) -> SyntaxChoice {
+        self.default_syntax
+    }
+
+    fn init(
+        default_syntax: SyntaxChoice,
+        pre_compiled_deps: Option<&'a FullyCompiledProgram>,
+        task_opt: Option<TaskInput<(InitCommand, Self::ExtraInitArgs)>>,
+    ) -> Self {
+        let additional_mapping = match task_opt.map(|t| t.command) {
+            Some((InitCommand { named_addresses }, _)) => {
+                verify_and_create_named_address_mapping(named_addresses).unwrap()
+            }
+            None => BTreeMap::new(),
+        };
+
+        let mut named_address_mapping = diem_framework::diem_framework_named_addresses();
+        for (name, addr) in additional_mapping {
+            if named_address_mapping.contains_key(&name) {
+                panic!(
+                    "Invalid init. The named address '{}' is reserved by either the move-stdlib or diem-framework",
+                    name
+                )
+            }
+            named_address_mapping.insert(name, addr);
+        }
+
+        // TODO: for now, we are using the FakeExecutor to keep things simple. This however,
+        // results in the genesis being compiled twice. We are planning to either move away
+        // from FakeExecutor in the future, or rework genesis generation.
+        Self {
+            compiled_state: CompiledState::new(named_address_mapping, pre_compiled_deps),
+            default_syntax,
+            exec: FakeExecutor::from_fresh_genesis(),
+        }
+    }
+
+    fn publish_module(
+        &mut self,
+        module: CompiledModule,
+        gas_budget: Option<u64>,
+        extra_args: Self::ExtraPublishArgs,
+    ) -> Result<()> {
+        let module_id = module.self_id();
+        let signer = module_id.address();
+        let params = get_default_transaction_parameters(&self.exec, signer);
+
+        let mut module_blob = vec![];
+        module.serialize(&mut module_blob).unwrap();
+
+        let txn = RawTransaction::new_module(
+            signer.clone(),
+            params.sequence_number,
+            TransactionModule::new(module_blob),
+            params.max_gas_amount,
+            params.gas_unit_price,
+            params.gas_currency_code,
+            params.expiration_timestamp_secs,
+            ChainId::test(),
+        )
+        .sign(
+            &extra_args.privkey,
+            Ed25519PublicKey::from(&extra_args.privkey),
+        )?
+        .into_inner();
+
+        // TODO: handle vm status
```

Comment:
Maybe make my status handler public, or utility-ify some things from the vm_test_harness

Reply:
I'll look into those options